servers/quarkus-common/src/main/java/org/projectnessie/quarkus/config/CatalogAdlsConfig.java

```diff
@@ -37,13 +38,29 @@
  * documentation</a>
  */
 @ConfigMapping(prefix = "nessie.catalog.service.adls")
-public interface CatalogAdlsConfig extends AdlsConfig, AdlsOptions<CatalogAdlsFileSystemOptions> {
+public interface CatalogAdlsConfig
+    extends AdlsConfig, AdlsOptions<CatalogAdlsFileSystemConfig, CatalogAdlsNamedFileSystemConfig> {
 
   /** Custom settings for the ADLS Java client. */
   @WithName("configuration")
   @Override
   Map<String, String> configurationOptions();
 
+  @Override
+  OptionalInt maxHttpConnections();
```

Comment:
Why these overrides?

Reply:
In general for each method in a `XYZOptions` there is an override in the corresponding `WYZConfig` (I think for readability reasons). I just added the missing overrides.

Reply:
Sorry but after thinking more about this, I went the opposite direction and removed all the useless overrides. The main reason is that it's already very hard to keep the config classes and the Helm chart in sync, and having overrides makes it even more complex, because you never know if all the methods are really overridden, so you need to go check the parent classes anyways.